cli/src/revset_util.rs

```diff
@@ -107,6 +107,31 @@ impl<'repo> RevsetExpressionEvaluator<'repo> {
     }
 }
 
+fn warn_user_redefined_builtin(
+    ui: &Ui,
+    source: &ConfigSource,
+    name: &str,
+) -> Result<(), CommandError> {
+    match source {
+        ConfigSource::Env | ConfigSource::User | ConfigSource::Repo | ConfigSource::CommandArg => {
+            let checked_mutability_builtins =
+                ["mutable()", "immutable()", "builtin_immutable_heads()"];
+
+            if checked_mutability_builtins.contains(&name) {
+                writeln!(
+                    ui.warning_default(),
+                    "redefining `revset-aliases.{}` is not recommended; redefine \
+                     `immutable_heads()` instead",
+                    name
```

Comment:
nit: capitalize "Redefining", and `name` can be inlined as `revset-aliases.{name}`.

Reply:
Done.